src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -399,17 +401,68 @@ internal long GetLength(ReadOnlySpan<char> path, bool continueOnError = false)
             return EntryExists ? _fileCache.Size : 0;
         }
 
+        internal UnixFileMode GetUnixFileMode(ReadOnlySpan<char> path, bool continueOnError = false)
+            => GetUnixFileMode(handle: null, path, continueOnError);
+
+        internal UnixFileMode GetUnixFileMode(SafeFileHandle handle, bool continueOnError = false)
+            => GetUnixFileMode(handle, handle.Path, continueOnError);
+
+        private UnixFileMode GetUnixFileMode(SafeFileHandle? handle, ReadOnlySpan<char> path, bool continueOnError = false)
+        {
+            EnsureCachesInitialized(handle, path, continueOnError);
+
+            if (!EntryExists)
+                return (UnixFileMode)(-1);
```

Comment:
Why not just throw directly from here? AFAICT, if you have a `FileSystemInfo` for a non-existent file and get `UnixFileMode`, you'll get `-1`, which I don't think is intended behavior.

Reply:
`FileSystemInfo` properties don't throw on get, they all return some constant value.

The `File.Get*Time*` methods do this also.

`File.GetAttributes` does throw. And I've implemented `File.GetUnixFileMode` to throw too.

`FileSystemInfo.Attributes` returns `(FileAttributes)(-1)` and that made me use `(UnixFileMode)(-1)` as the return value for not-exists on Unix for `UnixFileMode`.

On Windows, `UnixFileMode` returns `(FileAttributes)(-1)` unconditionally. Here it means _unsupported_ instead of _doesn't exist_.

Reply:
Hmm, I guess being consistent with FileAttributes makes sense. It is weird to me, but it works.